src/engine/renderer/tr_image.cpp

```diff
@@ -1611,6 +1611,58 @@ static const imageExtToLoaderMap_t imageLoaders[] =
 
 static int                   numImageLoaders = ARRAY_LEN( imageLoaders );
 
+/*
+=================
+R_FindAltImage
+
+Finds an alternative image path, set the required prefix if needed,
+then returns the loader index.
+=================
+*/
+int R_FindAltImage( char *filename, const char **prefix ) {
```

Comment:
in fact the function takes filename as argument (in fact a basename, I edited this), then return a prefix string and an index to a given format (for extension/loader), but this is c++ and wen can't return two variable neither return string :grin:

Reply:
also the prefix is only set on rare use case (dds prefix like DarkPlaces/Doom3), it looks a bit costly to compute a string for every texture because of a use case no one unvanquished player will face.

Reply:
Also, when this function is only used to tell if a file with a given basename exists (like in DarkPlaces compat code), both the prefix and the final filepath are not used, would be useless to compute that string for nothing. The good thing is that setting a prefix is just setting a pointer to a `const char*`, processing is only done when people really wants to load the picture. And since this is done later in any way, no need to compute two times the same string at two memory places to only use one if them.